unit-tests/live/calib/test_calibrations_common.py

```diff
@@ -0,0 +1,123 @@
+# Except as expressly permitted by the Software license, no part of the Software may be reproduced, stored in a retrieval system, transmitted in any form, or distributed by any means without the express written consent of RealSense, Inc.
+
+#########################################################################################################################################
+##                   tests for OCC and Tare calib flows                                                                ##
+#########################################################################################################################################
+import sys
+import time
+import pyrealsense2 as rs
+from rspy import test, log
+
+# Constants for calibration
+CALIBRATION_TIMEOUT_SECONDS = 30
+OCC_TIMEOUT_MS = 9000
+TARE_TIMEOUT_MS = 10000
+FRAME_PROCESSING_TIMEOUT_MS = 5000
+HARDWARE_RESET_DELAY_SECONDS = 3
+
+def on_calib_cb(progress):
+    """Callback function for calibration progress reporting."""
+    pp = int(progress)
+    log.i('\r' + '*' * pp + ' ' * (99 - pp) + '*')
+    if pp == 100:
+        log.i("progress - ", pp)
+
+def calibration_main(host_assistance, occ_calib, json_config, ground_truth):
+    """
+    Main calibration function for both OCC and Tare calibrations.
+    
+    Args:
+        host_assistance (bool): Whether to use host assistance mode
+        occ_calib (bool): True for OCC calibration, False for Tare calibration
+        json_config (str): JSON configuration string
+        ground_truth (float): Ground truth value for Tare calibration (None for OCC)
+    
+    Returns:
+        bool: True if calibration was not skipped, False if was skipped
+        float: Health factor from calibration
+    """
+    config = rs.config()
+    pipeline = rs.pipeline()
+    pipeline_wrapper = rs.pipeline_wrapper(pipeline)
+    pipeline_profile = config.resolve(pipeline_wrapper)
+    auto_calibrated_device = rs.auto_calibrated_device(pipeline_profile.get_device())
+    if not auto_calibrated_device:
+        log.e("Failed to open device for calibration")
+        test.fail()
+        return True, 0
+    
+    ctx = rs.context()
+    try:
+        device = ctx.query_devices()[0]
+    except IndexError:
+        log.e("Failed to connect device for calibration")
+        test.fail()
+        return True, 0
+    
+    # Skip test for MIPI devices and are not in host assistance mode
+    if not device.supports(rs.camera_info.usb_type_descriptor) and not host_assistance:
+        log.i("MIPI device - skip the test")
+        return False, 0
+    
+    # Bring device in start phase
+    device.hardware_reset()
+    time.sleep(HARDWARE_RESET_DELAY_SECONDS)
+
+    stream_config = rs.config()
+    if host_assistance:
+        stream_config.enable_stream(rs.stream.depth, 1280, 720, rs.format.z16, 30)            
```

Comment:
Why not use these resolutions when resolving above? Less efficient like this.

Reply:
Ahh... because you use hardware reset. See my previous comments about no need to reset and insufficient timeout.